app/services/force/listing_service.rb

```diff
@@ -53,7 +54,10 @@ def self.listing(id, opts = {})
       listing
     end
 
-    def self.translations_valid?(listing_translations)
+    def self.translations_valid?(listing_translations, listing_last_modified)
+      return false unless listing_translations[:LastModifiedDate] == listing_last_modified
```

Comment:
question: This seems to duplicate the method `CacheService#listing_unchanged?`, which gets run every 10 minutes via the `prefetch` job. Does this make the `prefetch` job redundant now? I.e. do we still need a background process to update caches for all modified listings every 10 minutes, when we are now doing it on every visit to a listing details page?

Reply:
This is a good question and is related to the open question in the description. 

This function compares the listing LastModifiedDate and the translations cache LastModifiedDate. 

This returns false when:
- A listing has been updated but the event subscriber failed to update the translations.

- With the force=true call in the google_translate_service, we should always refresh the listing object when we update translations. So in theory, the LastModifiedDate of the listing should be guaranteed to be accurate for the purposes of the comparison.

If this function is false, then we process translations -> force update the listing -> translate all fields.

So yes, if the event subscriber is working -> then everytime we update a listing -> we translate and force update the listing. We could probably remove the prefetch job in this case.

If the event subscriber is down for some reason, then we would not update translations or force refresh the listing. 
If we disable the prefetch job under this scenario: The listing wouldn't reflect the new changes until a manual force refresh. And we wouldn't catch the disagreement on the listing service call.

We could remove the force=true call from the google translate service, and then this check in the listing detail would return false due to the translations being newer. (Unless the prefetch job gets it first). The intended behavior would be to refresh the listing. Then the dates should match.

This ticket primarily aims to resolve the potential complications of the event subscriber failing. We do plan on implementing event replays so that is a future consideration. 

Hopefully this provides some insight into the potential paths forward. We could keep the prefetch job and introduce some redundancy that would cover all of the potential scenarios.

We could remove the force=true call from the google_translation service, and just let the listing_service be responsible for updating the listing cache. Perhaps that would simplify things for us. 

Maybe we should make the LastModifiedDate comparison have two outcomes: if the listing date is older than the translations -> refresh the listing. If the translation date is older than the listing -> refresh the translations.